app/scripts/controllers/metamask-notifications/types/on-chain-notification/on-chain-notification.ts

```diff
@@ -0,0 +1,50 @@
+import type { TRIGGER_TYPES } from '../../constants/notification-schema';
+import type { Compute } from '../type-utils';
+import type { components } from './schema';
+
+export type Data_MetamaskSwapCompleted =
+  components['schemas']['Data_MetamaskSwapCompleted'];
+export type Data_LidoStakeReadyToBeWithdrawn =
+  components['schemas']['Data_LidoStakeReadyToBeWithdrawn'];
+export type Data_LidoStakeCompleted =
+  components['schemas']['Data_LidoStakeCompleted'];
+export type Data_LidoWithdrawalRequested =
+  components['schemas']['Data_LidoWithdrawalRequested'];
+export type Data_LidoWithdrawalCompleted =
+  components['schemas']['Data_LidoWithdrawalCompleted'];
+export type Data_RocketPoolStakeCompleted =
+  components['schemas']['Data_RocketPoolStakeCompleted'];
+export type Data_RocketPoolUnstakeCompleted =
+  components['schemas']['Data_RocketPoolUnstakeCompleted'];
+export type Data_ETHSent = components['schemas']['Data_ETHSent'];
+export type Data_ETHReceived = components['schemas']['Data_ETHReceived'];
+export type Data_ERC20Sent = components['schemas']['Data_ERC20Sent'];
+export type Data_ERC20Received = components['schemas']['Data_ERC20Received'];
+export type Data_ERC721Sent = components['schemas']['Data_ERC721Sent'];
+export type Data_ERC721Received = components['schemas']['Data_ERC721Received'];
+
+type Notification = components['schemas']['Notification'];
+type NotificationDataKinds = NonNullable<Notification['data']>['kind'];
+type ConvertToEnum<Kind> = {
+  [K in TRIGGER_TYPES]: Kind extends `${K}` ? K : never;
+}[TRIGGER_TYPES];
+
+/**
+ * Type-Foo.
```

Comment:
Does `Type-Foo` need to be updated?

Reply:
LOL I meant `Type-Fu` (like kung fu) 😛 

I'll rename this comment